src/worldcereal/rdm_api/rdm_interaction.py

```diff
@@ -95,70 +104,108 @@ def _get_headers(self) -> Dict[str, str]:
+
         Returns
         -------
-        List[str]
-            A List containing the URLs of all intersection collection IDs.
+        List[RdmCollection]
+            A List containing the collections matching the criteria.
         """
 
-        bbox = geometry.bounds
+        # If user requests private collections, they must be authenticated
+        if include_private:
+            if not self.headers or "Authorization" not in self.headers:
+                logger.info("To access private collections, you need to authenticate.")
+                self.authenticate()
+
+        # Handle geometry
+        bbox = geometry.bounds if geometry is not None else [-180, -90, 180, 90]
+        # check if the geometry is valid
+        if bbox[0] < -180 or bbox[1] < -90 or bbox[2] > 180 or bbox[3] > 90:
+            raise ValueError("Invalid geometry. CRS should be EPSG:4326.")
         bbox_str = f"Bbox={bbox[0]}&Bbox={bbox[1]}&Bbox={bbox[2]}&Bbox={bbox[3]}"
 
+        # Handle temporal extent
         val_time = (
             f"&ValidityTime.Start={temporal_extent[0]}T00%3A00%3A00Z&ValidityTime.End={temporal_extent[1]}T00%3A00%3A00Z"
-            if temporal_extent
+            if temporal_extent is not None
             else ""
         )
 
-        url = f"{self.RDM_ENDPOINT}/collections/search?{bbox_str}{val_time}"
+        # Handle EWOC codes
+        ewoc_codes_str = (
+            "".join([f"&EwocCodes={str(ewoc_code)}" for ewoc_code in ewoc_codes])
+            if ewoc_codes is not None
+            else ""
+        )
+
+        # Construct the URL
+        url = f"{self.RDM_ENDPOINT}/collections/search?{bbox_str}{val_time}{ewoc_codes_str}"
 
+        # Process the request
         response = self.session.get(url=url, headers=self._get_headers(), timeout=10)
 
         if response.status_code != 200:
             raise Exception(f"Error fetching collections: {response.text}")
 
-        response_json = response.json()
+        collections = [RdmCollection(**col) for col in response.json()]
 
-        col_ids = [col["collectionId"] for col in response_json]
+        # Filter out public and private collections if needed
+        if not include_public:
+            collections = [col for col in collections if col.access_type != "Public"]
 
-        if not col_ids:
-            raise NoIntersectingCollections(
-                f"No spatiotemporally intersecting collection IDs found in the RDM for the given geometry: {bbox} and temporal extent: {temporal_extent}."
-            )
+        if not include_private:
+            collections = [col for col in collections if col.access_type == "Public"]
+
+        if not collections:
+            logger.info("No collections found in the RDM for your search criteria.")
 
-        return col_ids
+        return collections
 
     def _get_download_urls(
-        self, collection_ids: List[str], user_id: Optional[str] = None
+        self,
+        ref_ids: List[str],
```

Comment:
why is this called `ref_ids` if this about collection IDs? Confusing naming compared to our internal `ref_id`